Makefile

```diff
@@ -50,6 +50,42 @@ ifndef OPTIM
     OPTIM := -O3
 endif
 
+ifdef HOST
+    ifeq "$(CC)" "cc"
+        CC = $(HOST)-gcc
+    endif
+    ifeq "$(CXX)" "c++"
+        CXX = $(HOST)-g++
+    endif
+    ifeq "$(CPP)" "cpp"
+        CPP = $(HOST)-cpp
+    endif
+    ifeq "$(LD)" "ld"
+        LD = $(HOST)-gcc
+    endif
+    ifeq "$(AR)" "ar"
+        AR = $(HOST)-ar
+    endif
+    ifeq "$(AS)" "as"
+        AS = $(HOST)-as
+    endif
+endif
+
+ifdef RUNTIME
+    ifeq "$(RUNTIME)" "FreeRTOS-lwIP"
+        ifndef FREERTOS_TOP
+            $(error FREERTOS_TOP not supplied with RUNTIME=$(RUNTIME))
```

Comment:
Getting this: 
```
make HOST=arm-none-eabi RUNTIME=FreeRTOS-lwIP
Makefile:77: *** FREERTOS_TOP not supplied with RUNTIME=FreeRTOS-lwIP.  Stop.
```
But don't see `FREERTOS_TOP` mentioned in PR comments or in any files.

Reply:
will fix.  I had sent some out-of-band build instructions, ```make -j HOST=arm-none-eabi RUNTIME=FreeRTOS-lwIP FREERTOS_TOP=../third/FreeRTOSv202212.00/FreeRTOS/Source LWIP_TOP=../third/lwip/src EXTRA_CFLAGS='-mthumb -mcpu=cortex-m7'``` but that needs to make it into the distribution.  I'll add documentation around it in README.md, and I'm also adding a subtest to `make check` to build+unittest under arm-none-eabi-FreeRTOS-lwIP.